internal/pkg/crypto/ca.go

```diff
@@ -18,28 +18,28 @@ type CA struct {
 	CaPrivateKey *rsa.PrivateKey
 }
 
-func (ca *CA) PEMEncodeCertificate() (string, error) {
+func (ca *CA) PEMEncodeCertificate() (*bytes.Buffer, error) {
 	caPEM := new(bytes.Buffer)
 	err := pem.Encode(caPEM, &pem.Block{
 		Type:  "CERTIFICATE",
 		Bytes: ca.CaCertBytes,
 	})
 	if err != nil {
-		return "", errors.Join(errors.New("cannot PEM encode certificate"), err)
+		return nil, errors.Join(errors.New("cannot PEM encode certificate"), err)
 	}
-	return caPEM.String(), nil
+	return caPEM, nil
 }
 
-func (ca *CA) PEMEncodePrivateKey() (string, error) {
+func (ca *CA) PEMEncodePrivateKey() (*bytes.Buffer, error) {
```

Comment:
Same here, is there a reason to not return a `string`?

Reply:
To void another conversion. This function is returning `*bytes.Buffer` because I do not need string value. All the usage of the data returned by this function needs to be bytes. Not strings. But, I keep using `*bytes.Buffer` instead of `[]byte` because if in the next changes is necessary a string I could access it using the available function [`String`](https://pkg.go.dev/bytes@go1.21.4#Buffer.String) or [`Bytes`](https://pkg.go.dev/bytes@go1.21.4#Buffer.Bytes).